external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -623,7 +624,7 @@ class AvroSuite extends SparkFunSuite {
       spark.read.avro("*/*/*/*/*/*/*/something.avro")
     }
 
-    intercept[FileNotFoundException] {
+    intercept[java.io.IOException] {
       TestUtils.withTempDir { dir =>
         FileUtils.touch(new File(dir, "test"))
         spark.read.avro(dir.toString)
```

Comment:
We can fix the case as
```
spark.read.option("avro.mapred.ignore.inputs.without.extension", false).avro(dir.toString)
```
The behavior will be the same as before. And we don't need to modify the expected `FileNotFoundException`

Reply:
I would actually remove this piece of code from the test, and write a separate test (reading a folder with files without `.avro` extensions) in which the Hadoop's parameter `avro.mapred.ignore.inputs.without.extension` is set to `true` explicitly. Unfortunatelly there is no method like `withSQLConf` for Hadoop's configs.